examples/obstacle_circumnavigation/common.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+// BoB robotics includes
+#include "common/pose.h"
+#include "navigation/read_objects.h"
+#include "robots/control/collision_detector.h"
+#include "robots/control/obstacle_circumnavigation.h"
+#include "robots/tank.h"
+#include "viz/sfml_world/arena_object.h"
+#include "viz/sfml_world/sfml_world.h"
+
+// Eigen
+#include <Eigen/Core>
+
+// SFML
+#include <SFML/Graphics.hpp>
+
+// Standard C++ includes
+#include <chrono>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+using namespace BoBRobotics;
+using namespace units::angle;
+using namespace units::literals;
+using namespace units::length;
+using namespace std::literals;
+
+struct Noop
+{
+    void operator()()
+    {}
+};
+
+template<class PoseGetterType, class Func = Noop>
+void
+runObstacleCircumnavigation(Robots::Tank &tank, PoseGetterType &poseGetter, Func extraCalls = Noop())
```

Comment:
I don't really like this ``extraCalls`` motif - I'd just stick it in a class with most of the stuff in the constructor and then have an update function which the user can run in whatever sort of loop they want

Reply:
I agree this is a bit of a nasty hack. I'll take a look and see if I can make a nicer API...

Reply:
I've replaced this with a class that has an ``update()`` method that can be polled instead, which does make it a lot less gross.